pkg/api/v1/resource/helpers.go

```diff
@@ -83,6 +85,11 @@ func PodRequests(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 		addResourceList(reqs, containerReqs)
 	}
 
+	var sidecarResources v1.ResourceList
+	if opts.SidecarContainersEnabled {
+		sidecarResources = v1.ResourceList{}
+	}
```

Comment:
```suggestion
	var sidecarResources v1.ResourceList
	if opts.SidecarContainersEnabled {
		sidecarResources = make(v1.ResourceList)
	}
```

nit:

`s/v1.ResourceList{}/make(v1.ResourceList)/g`

for readability? (It might look a bit weird about why this is needed if someone doesn't know v1.ResourceList is actually a map)